x-pack/test/functional_gen_ai/inference/tests/index.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FtrProviderContext } from '../ftr_provider_context';
+import { getAvailableConnectors } from '../../common/connectors';
+
+// eslint-disable-next-line import/no-default-export
+export default function ({ loadTestFile, getService }: FtrProviderContext) {
+  describe('Inference plugin - API integration tests', async () => {
+    const currentConnector = getService('currentConnector');
+
+    getAvailableConnectors().forEach((connector) => {
+      describe(`Connector ${connector.id}`, () => {
+        before(() => {
+          currentConnector.set(connector.id);
+        });
+
+        loadTestFile(require.resolve('./chat_complete'));
+      });
+    });
```

Comment:
I wanted to re-use the exact same test files for each connector, so I went with this `currentConnector` get/set service. It work-ish for now, but given services can only be accessed in lifecycle blocks (either tests or before/after hooks), it might become an issue if we want to, say not running a specific test for a specific connector.. I might need to use a "test factory" pattern as we're doing elsewhere then...